camp/camp-brooklyn/src/test/java/org/apache/brooklyn/camp/brooklyn/ConfigNestedYamlTest.java

```diff
@@ -0,0 +1,148 @@
+        Entities.dumpInfo(ent);
+    }
+
+    @Test
+    public void testCatalogParameterFromSuperYamlTypeAfterTemplateAccess() throws Exception {
+        addCatalogItems( loadYaml("config-nested-test.bom") );
+        Entity ent = makeBlueprint( "services: [ { type: test-map-parameter } ]");
+        
+        checkEntity(ent, false);
+    }
+
+    protected void checkTemplates(EntityInternal ent) {
+        Map<String, Object> substitutions = MutableMap.copyOf(ent.config().get(SoftwareProcess.TEMPLATE_SUBSTITUTIONS)).asUnmodifiable();
+        String out0 = TemplateProcessor.processTemplateContents("hello "+"${field}", substitutions);
+        Assert.assertEquals(out0, "hello val");
+        
+        String out1 = TemplateProcessor.processTemplateContents("hello "+"${config['template.substitutions']['field']}", ent, substitutions);
+        Assert.assertEquals(out1, "hello val");
+    }
+
+    @Test
+    public void testCatalogParameterFromSuperYamlTypeInCluster() throws Exception {
+        addCatalogItems( loadYaml("config-nested-test.bom") );
+        Entity cluster = makeBlueprint("services: [ { type: test-cluster-with-map-parameter } ]");
+        Entities.dumpInfo(cluster.getApplication());
+        Entity parentInCluster = Iterables.getOnlyElement( ((DynamicCluster)cluster).getMembers() );
+        Entity target = Iterables.getOnlyElement(parentInCluster.getChildren());
+        checkEntity( target, false );
+    }
+    
+
+    @Test
+    public void testCatalogParameterFromSuperYamlTypeAsSoftware() throws Exception {
+        addCatalogItems( loadYaml("config-nested-test.bom") );
+        Entity ent = doTestWithBlueprint( "services: [ { type: test-map-parameter-software } ]", true);
+        Entities.dumpInfo(ent);
+    }
+
+    @Test
+    public void testCatalogNoParameterAsSoftware() throws Exception {
+        addCatalogItems( loadYaml("config-nested-test.bom") );
+        doTestWithBlueprint( "services: [ { type: test-no-parameter-software } ]", true);
+    }
+
+    protected Entity doTestWithBlueprint(String bp, boolean expectDotSyntaxToWork) throws Exception {
+        return checkEntity(makeBlueprint(bp), expectDotSyntaxToWork);
+    }
+    
+    protected Entity makeBlueprint(String bp) throws Exception {
+        Entity app = createAndStartApplication( bp );
+        waitForApplicationTasks(app);
+        return Iterables.getOnlyElement( app.getChildren() );
+    }
+    
+    protected Entity checkEntity(Entity ent, boolean expectDotSyntaxToWork) throws Exception {
+        String dotSyntax = ent.getConfig( ConfigKeys.newStringConfigKey("template.substitutions.field") );
+        Assert.assertEquals(dotSyntax, expectDotSyntaxToWork ? "val" : null, "wrong value for dot syntax: "+dotSyntax);
+        // observed in some situations we get the dot syntax instead!
+        // Assert.assertEquals(dotSyntax, "val", "wrong value for dot syntax: "+dotSyntax);
```

Comment:
I'd have expected this to always be the case, for any `Map` value, not just for `MapConfigKey`. The logic doing the lookup (if I remember correctly) is trying to get for increasingly longer parts of the full key until it has a hit.